internal/redispool/sysreq.go

```
@@ -20,14 +20,36 @@ func init() {
 
 func redisCheck(name, addr string, timeout time.Duration, pool *redis.Pool) sysreq.CheckFunc {
 	return func(ctx context.Context) (problem, fix string, err error) {
-		deadline := time.Now().Add(timeout)
+		check := func() (err error) {
+			// Instead of just a PING, we also use this hook point to force a rewrite of
+			// the AOF file on startup of the frontend as a way to ensure it doesn't
+			// grow out of bounds which slows down future startups.
+			// See https://github.com/sourcegraph/sourcegraph/issues/3300 for more context
 
-		for time.Now().Before(deadline) {
 			c := pool.Get()
-			_, err = c.Do("PING")
-			c.Close()
+			defer func() { _ = c.Close() }()
+
+			if err = c.Send("PING"); err != nil {
+				return err
+			}
 
-			if err == nil {
+			if err = c.Send("BGREWRITEAOF"); err != nil {
```

can we do this in a background goroutine rather? No need to prevent the starting of sourcegraph on this command. Maybe move it to to after we reach success on line 53 below, rather than mixing it in with the "check" logic. I see you document it as async, but still think it would be clearer to not mix it up in the check function.

---

> No need to prevent the starting of sourcegraph on this command

The error here is not from the execution of the command, but rather only an error that may happen when writing to the in-memory buffer, or a previous error that happened in the connection. We explicitly ignore any error or result from this command, because never call `Receive` for it (as written in a comment).

Given that, would you still want this changed?

---

so will it block at all? If not it should be fine.

---

It will not block.

---

do we still need this now with the other changes made?

---

Yes, because server still has frequent restarts and that means the BGREWRITEAOF operation isn't triggered enough. With the new setting, that'd mean we'd never actually write the RDB preamble, so it'd have no effect otherwise.